backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java

```diff
@@ -0,0 +1,38 @@
+package de.wordsmith.backend.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class PonsApiService {
+
+    private final String ponsApi = "https://api.pons.com/v1/dictionary?l=dede&q=";
+    private final RestTemplate restTemplate;
+    @Value("${secret}")
+    private String secret;
+
+    @Autowired
+    public PonsApiService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public boolean doesWordExist(String wordText) {
+        String requestString = ponsApi + wordText;
+        try {
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.set("X-Secret", secret);
+            HttpEntity httpEntity = new HttpEntity(httpHeaders);
+            ResponseEntity<Void> response = restTemplate.exchange(requestString, HttpMethod.GET, httpEntity, Void.class);
+            HttpStatus statusCode = response.getStatusCode();
+            return statusCode.equals(HttpStatus.OK);
+        } catch (HttpClientErrorException e) {
+            System.out.println(e.getMessage());
+            return false;
```

Comment:
simply returning `false` here may be misleading, as the word might still exist but e.g. the API is not responsive at the moment - you may want to consider to pass an error to your frontend allowing you to advise your user to try adding the word at a later time

Reply:
Already planned as the next task but thanks for the suggestion :)